src/treelearner/leaf_splits.hpp

```diff
@@ -22,8 +22,13 @@ namespace LightGBM {
 class LeafSplits {
  public:
   LeafSplits(data_size_t num_data, const Config* config)
-    :config_(config), num_data_in_leaf_(num_data), num_data_(num_data),
+      : deterministic_(false),
```

Comment:
/gha run r-valgrind

Reply:
looks like your changes fixed it!

https://github.com/microsoft/LightGBM/runs/1350825291?check_suite_focus=true

The only issues left there are the ones we already knew about as of #3443 , which we think are false positives.